Indicate bug step number in the left bug step overview tree

csordasmarton opened the pull request:
Closes #778

csordasmarton commented:
I made a screenshot from this:
![cc_bugstep](https://user-images.githubusercontent.com/6695818/29262410-b23efbde-80d4-11e7-8fcb-91a39d560d1a.png)

Xazax-hun commented:
I do like how this looks, however, I am not sure if red is a good choice of color here. My problem is, red tends to be the color of errors. In this case, `Assuming...` is not an error.

whisperity commented:
I feel that the numbers are too big, compared to the rest of the GUI.

csordasmarton commented:
@Xazax-hun I separated the colors of the bubbles to blue (information) and red (error - last item). What do you think about this?

@whisperity The numbers are bigger, because I added a bold style to it. I removed it.

![cc_bugstep_2](https://user-images.githubusercontent.com/6695818/29268374-c7202f74-80ec-11e7-8bd3-0c940f3c3938.png)

Xazax-hun commented:
Looks good, thank you!

whisperity commented:
`fixit`s contain a special [` (fixit)` marker](https://github.com/Ericsson/codechecker/blob/7d031bfa4241d277ac6b3e18116656fcb4dfa031/libcodechecker/analyze/tidy_output_converter.py#L337) at the end of their string. Maybe we should use a different highlight (yellow? orange? green?... I'd vouch for yellow.) for those?

Currently it looks like this:
![image](https://user-images.githubusercontent.com/1969470/29268955-64e2e2fe-80ef-11e7-88d7-cab9e4cd4b42.png)

@Xazax-hun, what do you think?

Xazax-hun commented:
@whisperity ,I agree, good idea.

whisperity commented:
As a final tweak what I would like to see, if possible, is **not** showing the bug number if the entire bug is consistent of a single line (e.g. `deadcode.DeadStores`).

Consider that if there aren't any steps, only one report is shown in the tree view (and that has no number), so I believe the marker in the code viewer shouldn't have a number marker either.

![image](https://user-images.githubusercontent.com/1969470/29274311-b31e2e3e-8107-11e7-858c-e639a361ad08.png)

Other than this, this looks lovely!

whisperity commented on www/scripts/codecheckerviewer/BugViewer.js:
```diff
@@ -78,6 +78,31 @@ function (declare, domClass, dom, style, fx, Toggler, on, query, Memory,
+
+    if (message.indexOf('(fixit)') > -1)
+      ret = '<span class="tag fixit">fixit</span>'
+        + message.replace('(fixit)', '');
```
The text is appended as ` (fixit)` with a space before the tag. While this is a very minor stuff, I think we should stript in entirely.
